src/main/java/de/upb/crypto/math/structures/groups/lazy/LazyGroup.java

```diff
@@ -43,6 +41,16 @@
     boolean isPrimeOrder;
     Zn zn;
     GroupElement generator;
+    /**
+     * We use wNAF by default but you can enable sliding window multi-exponentiation algorithm by setting this to
+     * {@code true}. Keep in mind that for negative exponents, the precomputation is not cached.
+     */
+    boolean useSlidingWindowMultiexp;
+    /**
+     * We use wNAF by default but you can enable sliding window exponentiation algorithm by setting this to
+     * {@code true}. Keep in mind that for negative exponents, the precomputation is not cached.
+     */
+    boolean useSlidingWindowExp;
```

Comment:
Can you make this an `enum`, please? I see no reason to wait.

Reply:
Just for the multi-exponentiation selector, right? There is no simultaneous algorithm for the single exponentiation. Unless we want to allow the naive binary square-and-multiply algorithm too.

Reply:
I'd have said for single exponentiation, we also have choices. For example, one can choose between Wnaf and sliding window (ideally depending on whether or not inversion is cheap). 
So I guess for completeness and consistency, both multi- and single-exponentation algorithms should be choosable? Unless I'm missing something

Reply:
We already have the boolean to do that. An enum is only necessary if we have more than two options.

Reply:
I mean ... technically - of course! But there are more than two options in nature and potentially at some point we'll implement some of those. When that happens, I don't want to have to do a new major release to break the "boolean" API.
If we introduce the enum(s) now, we can later add options to it without breaking API.

Reply:
Okay, I am changing both to use enums.